lizrd/scripts/grid.py

```diff
@@ -6,244 +6,192 @@
-        raise ValueError("Singularity image is not specified (in JSON or env variable)")
-
-    if NODELIST is not None:
-        NODELIST = "--nodelist=" + NODELIST
-
-    grid = create_grid(PARAMS)
-    grid = multiply_grid(grid, RUNS_MULTIPLIER)
-    no_experiments = len(grid)
-    minutes_per_exp = timestr_to_minutes(TIME)
-
-    if len(grid) > 1 and runner == MachineBackend.LOCAL and not DRY_RUN:
+
+    config["params"]["path_to_config"] = path
+
+    interactive_debug_session = config.get("interactive_debug", False)
+
+    # list of pairs: a dictionary of training_args and a dictionary of setup_args
+    grid = []
+    total_n_experiments = 0
+    total_minutes = 0
+
+    pprint.pprint(config)
+    single_exp_training_args_grid = create_grid(config["params"])
+
+    setup_args = get_setup_args_with_defaults(config, CLUSTER_NAME)
+    single_exp_training_args_grid = multiply_grid(
+        single_exp_training_args_grid, setup_args["runs_multiplier"]
+    )
+    n_experiments = len(single_exp_training_args_grid)
+
+    grid += list(zip(single_exp_training_args_grid, [setup_args] * n_experiments))
+
+    total_n_experiments += n_experiments
+    minutes_per_exp = timestr_to_minutes(setup_args["time"])
+    total_minutes_from_this_grid = n_experiments * minutes_per_exp
+    total_minutes += total_minutes_from_this_grid
+
+    if CLUSTER_NAME == MachineBackend.LOCAL and len(grid) > 1:
         raise ValueError(
             f"Running more than one experiment locally is not supported (you are trying to run {len(grid)} experiments). Aborting..."
         )
 
-    total_minutes = no_experiments * minutes_per_exp
-    if not runner == MachineBackend.LOCAL:
-        if not INTERACTIVE_DEBUG:
+    if not CLUSTER_NAME == MachineBackend.LOCAL:
+        if not interactive_debug_session:
             user_input = input(
-                f"Will run {no_experiments} experiments, using up {total_minutes} minutes, i.e. around {round(total_minutes / 60)} hours"
-                f"\nExperiment settings: \n{RUNNER=} \n{TIME=} \n{N_GPUS=} \nContinue? [Y/n] "
+                f"Will run {total_n_experiments} experiments, using up {total_minutes} minutes, i.e. around {round(total_minutes / 60)} hours\n"
+                f"Continue? [Y/n]"
             )
         else:
-            user_input = input(f"Will run an INTERACTIVE experiment. \nContinue? [Y/n]")
+            user_input = input(
+                f"Will run an INTERACTIVE experiment, which will be the first one from the supplied configs. \nContinue? [Y/n]"
+            )
         if user_input.lower() not in ("", "y", "Y"):
             print("Aborting...")
             exit(1)
 
-    slurm_command = "srun" if INTERACTIVE_DEBUG else "sbatch"
-
-    if not (INTERACTIVE_DEBUG or runner == MachineBackend.LOCAL):
-        exp_name = next(iter(grid))["name"]
+    if not (interactive_debug_session or CLUSTER_NAME == MachineBackend.LOCAL):
+        exp_name = grid[0][0]["name"]
```

Comment:
Na pierwszy rzut oka nie wiadomo o co chodzi.
Doczytałem, że grid to lista (training_args, setup_args)

```suggestion
        first_exp_training_args, _ = grid[0]
        exp_name = first_exp_training_args["name"]
```